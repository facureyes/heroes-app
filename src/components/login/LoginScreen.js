import React, {useContext} from 'react'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';
// import { authReducer } from '../auth/authReducer'


const LoginScreen = ({ history }) => {
  
  const { dispatch } = useContext(AuthContext);

  const handleClick = () => {
    // Push add a new path to history and let user go back
    // history.push('/');
    // Replace as it name indicates, replaces the actual path in history, so user wont be able to navigate back
    // history.replace('/') ;
    dispatch({
      type: types.login,
      payload: {
        name: 'Facu',
        logged: true,
      }
    })
    history.replace('/') ;
  }
  return (
    <div className="container mt-5">
      <h1>Login Screen</h1>
      <hr />
      <button
        className="btn btn-dark"
        onClick={handleClick}
      >
        Login
      </button>
    </div>
  )
}

export default LoginScreen
