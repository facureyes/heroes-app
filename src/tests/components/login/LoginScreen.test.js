import { mount } from "enzyme"
import { MemoryRouter } from "react-router-dom"
import { AuthContext } from "../../../auth/AuthContext"
import LoginScreen from "../../../components/login/LoginScreen"

describe('Tests on <LoginScreen />', () => {
  
  const mockHistory={
    replace: jest.fn(),
  }

  const {user, dispatch} = {
    user: {
      logged: false,
    },
    dispatch: jest.fn(),
  }

  const wrapper = mount(
  <AuthContext.Provider value={{ dispatch }}>
      <LoginScreen history={mockHistory}/>
  </AuthContext.Provider>
  );

  test('should show properly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should make the dispatch and nav', () => {
    wrapper.find('button').simulate('click')

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(mockHistory.replace).toHaveBeenCalledTimes(1);
  })
  
  
})
