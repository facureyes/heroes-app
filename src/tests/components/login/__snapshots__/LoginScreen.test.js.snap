// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <LoginScreen /> should show properly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-dark"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
